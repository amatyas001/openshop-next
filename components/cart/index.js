import { CartContent } from './CartContent/CartContent';
import { CartAmount } from './CartAmount/CartAmount';
import { CartButton } from './CartButton/CartButton';
import { CartItem } from './CartItem/CartItem';
import { CartPanel } from './CartPanel/CartPanel';

export { CartContent, CartAmount, CartButton, CartItem, CartPanel };
