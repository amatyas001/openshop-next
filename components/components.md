Elements are composed by [Chakra-UI](https://chakra-ui.com) components and preserve the library features so [Style Props](https://chakra-ui.com/style-props) can be passed to them and accessibility remains optimized.

> _Style Props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components._

For more information visit the [Chakra UI](https://chakra-ui.com) documentation.

Components are divided into two main category: _wrappers_ and _single responsibility_ elements. Wrapper components does not contain business logic, however there are special cases like _handling lifted states to keep sensitive data locally_ and not to populate them to the Redux store which is persisted on the machine.

**State Dependencies** and **Wrapped Components** are documented in each description to give brief information about the context of posible interactions of the actual component. _Examples_ prepared with proper state and may render extra controls to test different cases with ease.
