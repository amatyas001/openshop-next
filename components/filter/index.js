export * from './FilterButton/FilterButton';
export * from './FilterColor/FilterColor';
export * from './FilterPrice/FilterPrice';
export * from './FilterName/FilterName';
export * from './FilterPanel/FilterPanel';
export * from './FilterReset/FilterReset';
