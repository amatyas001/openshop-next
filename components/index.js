export * from './shared';
export * from './cart';
export * from './payment';
