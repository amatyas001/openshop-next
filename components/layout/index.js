export * from './Content/Content';
export * from './Footer/Footer';
export * from './ProductsTopBar/ProductsTopBar';
export * from './Navbar/Navbar';
export * from './ScrollTop/ScrollTop';
