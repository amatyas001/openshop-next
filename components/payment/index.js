export * from './PaymentCancelled/PaymentCancelled';
export * from './PaymentConfirm/PaymentConfirm';
export * from './PaymentConfirmCard/PaymentConfirmCard';
export * from './PaymentConfirmControls/PaymentConfirmControls';
export * from './PaymentConfirmDetails/PaymentConfirmDetails';
export * from './PaymentError/PaymentError';
export * from './PaymentForm/PaymentForm';
export * from './PaymentFormControls/PaymentFormControls';
export * from './PaymentProgress/PaymentProgress';
export * from './PaymentReview/PaymentReview';
export * from './PaymentSuccess/PaymentSuccess';
