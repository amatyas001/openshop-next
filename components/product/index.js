import { ProductCard } from './ProductCard/ProductCard';
import { ProductControls } from './ProductControls/ProductControls';
import { ProductControlsColor } from './ProductControlsColor/ProductControlsColor';
import { ProductControlsAmount } from './ProductControlsAmount/ProductControlsAmount';
import { ProductControlsButton } from './ProductControlsButton/ProductControlsButton';
import { ProductControlsSize } from './ProductControlsSize/ProductControlsSize';

export {
  ProductCard,
  ProductControls,
  ProductControlsColor,
  ProductControlsAmount,
  ProductControlsButton,
  ProductControlsSize,
};
