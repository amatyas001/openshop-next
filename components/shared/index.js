import { Button } from './Button/Button';
import { ButtonIcon } from './ButtonIcon/ButtonIcon';
import { Field } from './Field/Field';
import { Image } from './Image/Image';
import { Panel } from './Panel/Panel';
import { Spinner } from './Spinner/Spinner';

export { Button, ButtonIcon, Field, Image, Panel, Spinner };
