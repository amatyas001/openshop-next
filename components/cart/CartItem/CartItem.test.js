import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { INITIAL_STATE } from '@app/config/';
import { mockProductWithBuyAmount } from '@app/mocks';
import { removeFromCart } from '@app/lib/redux/actions';
import { CartItem } from '@app/components';

let tree;
let store;
const mockStore = configureStore([]);
const product = mockProductWithBuyAmount(1)[0];

describe('<CartItem />', () => {
  beforeAll(() => {
    store = mockStore(INITIAL_STATE);
    act(() => {
      tree = create(
        <Provider store={store}>
          <CartItem item={product} />
        </Provider>
      );
    });
  });

  it('should render with required props', () => {
    expect.assertions(1);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should remove item from cart', () => {
    act(() => {
      tree = create(
        <Provider store={store}>
          <CartItem item={product} icons />
        </Provider>
      );
    });
    act(() => {
      tree.root
        .findByProps({
          'data-testid': 'cart-item-button-delete',
        })
        .props.handler();
    });
    expect.assertions(1);
    expect(store.getActions()).toEqual([removeFromCart(product)]);
  });
});
