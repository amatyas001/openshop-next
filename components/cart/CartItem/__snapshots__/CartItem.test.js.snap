// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> should render with required props 1`] = `
<div
  className="css-1tp0oa"
>
  <img
    alt="mock_name_0"
    className="css-rhemcv"
  />
  <div
    className="css-zbq0po"
  >
    <strong
      className="css-airqvi"
    >
      mock_name_0
    </strong>
    <em
      className="css-1fji79j"
    >
      mock_color_0 in One Size x1
    </em>
  </div>
  <strong
    className="css-1som8bm"
  >
    1.00
    Â $
  </strong>
</div>
`;
